pkg/evmclient/client.go

```diff
@@ -99,6 +101,31 @@ func DialContext(ctx context.Context, rawurl string, httpClient *http.Client) (*
 	return client, nil
 }
 
+func DialContextWithTimeout(ctx context.Context, rawurl string, httpClient *http.Client, timeout time.Duration) (*Client, error) {
```

Comment:
I think we can use `context.WithTimeout` to set timeout when connecting to node:

```go
ctx, cancel := context.WithTimeout(ctx, timeout)
return DialContext(ctx, rawurl, httpClient)
```

Reply:
Since I don't see any context handler inside `DialContext` function, I write a separate `DialContextWithTimeout` to handle the context (`case <-ctx.Done()`).

I conducted a timeout test and your suggestion worked. It seems like the RPC libraries handle context within them. However, I would rather manually handle the context than rely on libraries.

On the other hand, your suggestion is much cleaner.